scripts/world_bank/worldbank.py

```diff
@@ -47,222 +47,347 @@
+            # Filter the dataframe by this observation.
+            codes_that_match = codes_that_match.query(
+                f"{column} == {include_col}")
+            # Include the column in TMCF and column list.
+            if include_col:
+                base_template_mcf = (base_template_mcf
+                    + f"{column}: C:WorldBank->{column}\n")
+                cols_to_include_in_csv.append(f"{column}")
+
+        tmcfs_for_stat_vars.append((base_template_mcf, cols_to_include_in_csv,
+            list(indicator_codes.loc[codes_that_match.index]['IndicatorCode'])))
+
+    return tmcfs_for_stat_vars
 
-    Returns:
-      Array of tuples of templace MCF, columns on stat var observations, indicator codes for that template MCF
-  """
-
-  # Group stat vars by inclusion of variable in stat var observation
-  properties_belonging_to_stat_var_observation = ['measurementMethod', 'unit']
-  
-  tmcfs_for_stat_vars = []
-  null_status = indicator_codes.notna()
-
-  for permutation in list(itertools.product([False,True],repeat=len(properties_belonging_to_stat_var_observation))):
-    codes_that_match = null_status.copy()
-    base_template_mcf = TEMPLATE_TMCF
-    cols_to_include_in_csv = ['IndicatorCode']
-
-    for include, column in zip(permutation, properties_belonging_to_stat_var_observation):
-      codes_that_match = codes_that_match.query(f"{column} == {include}")
-
-      if include:
-        base_template_mcf = base_template_mcf + f"{column}: C:WorldBank->{column}"
-        cols_to_include_in_csv.append(f"{column}")
-
-    tmcfs_for_stat_vars.append((base_template_mcf, cols_to_include_in_csv, list(indicator_codes.loc[codes_that_match.index]['INDICATOR_CODE'])))
-
-  return tmcfs_for_stat_vars
 
 def download_indicator_data(worldbank_countries, indicator_codes):
-  """ Downloads world bank country data for all countries in provided dataframe. Retains only the unique indicator codes provided.
-  Additionally, creates a mapping from isoCode to MCF node.
+    """ Downloads world bank country data for all countries and
+            indicators provided.
+
+        Retains only the unique indicator codes provided.
+
+        Args:
+            worldbank_countries: Dataframe with ISO_CODE for each country.
+            indicator_code: Dataframe with INDICATOR_CODES to include.
+
+        Returns:
+            worldbank_dataframe: A tidied pandas dataframe where each row has
+            the format (indicator code, country ISO, year, value)
+            for all countries and all indicators provided.
+    """
+    worldbank_dataframe = pd.DataFrame()
+    indicators_to_keep = list(indicator_codes['IndicatorCode'].unique())
+
+    for index, country_code in enumerate(worldbank_countries['CountryCode']):
+        print(f"Downloading {country_code}")
+        country_df = read_worldbank(country_code)
+
+        # Remove unneccessary indicators.
+        country_df = country_df[country_df['IndicatorCode']
```

Comment:
I do not think that read_worldbank guarantees the returned data frame is not empty so it may have no columns.